Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -0,0 +1,462 @@
+#include <XR.h>
+
+#import <UIKit/UIKit.h>
+#import <ARKit/ARKit.h>
+#import <ARKit/ARConfiguration.h>
+
+extern void* GCALayerPtr;
+
+extern id<MTLDevice> MetalDevice;
+@interface SessionDelegate : NSObject <ARSessionDelegate>
+@property (readonly) id<MTLTexture> cameraTexture;
+@end
+
+@implementation SessionDelegate
+{
+    int width;
```

Comment:
Can we use universal initializers here? Not sure if objective C somehow interferes with this.
`int width{};`

Reply:
Nop because of ObjC